File: app/src/main/java/com/example/oops/Story1.java
package com.example.oops;

public class Story1 {
    public static void main(String[] args) {
        Employee e1 = new Contracter("Shashwat", "Contractor", 1);
        Employee e2 = new Intern("Shashwat", "Intern", 2);
        Employee e3 = new FullTime("Shashwat", "FullTime", 3);
        System.out.println(e1);
        System.out.println(e2);
        System.out.println(e3);
    }

}

class Employee {
    String employeeName;
    String employeeType;
    int employeeId;

    Employee(String employeeName, String employeeType,int employeeId){
        this.employeeId = employeeId;
        this.employeeName = employeeName;
        this.employeeType = employeeType;
    }

    public String toString(){
        return "Name: "+employeeName+" Type: "+employeeType+" ID: "+employeeId;
    }
}

class Contracter extends Employee{
    String salary = "$100,000";
    Contracter(String employeeName, String employeeType, int employeeId) {
        super(employeeName, employeeType, employeeId);
    }

    public String toString(){
        return super.toString() + " Salary: "+salary;
    }

}

class Intern extends Employee{

    String salary = "$50,000";

    Intern(String employeeName, String employeeType, int employeeId) {
        super(employeeName, employeeType, employeeId);
    }

    public String toString(){
        return super.toString() + " Salary: "+salary;
    }
}

class FullTime extends Employee{

    String salary = "$200,000";
    FullTime(String employeeName, String employeeType, int employeeId) {
        super(employeeName, employeeType, employeeId);
    }

    public String toString(){
        return super.toString() + " Salary: "+salary;
    }
}

File: app/src/main/java/com/example/oops/Story1.kt
package com.example.oops

object Story1 {
    @JvmStatic
    fun main(args: Array<String>) {
        val e1: Employee = Contracter("Shashwat", "Contractor", 1)
        val e2: Employee = Intern("Shashwat", "Intern", 2)
        val e3: Employee = FullTime("Shashwat", "FullTime", 3)
        println(e1)
        println(e2)
        println(e3)
    }
}

open class Employee(
    var employeeName: String,
    var employeeType: String,
    var employeeId: Int,
) {
    override fun toString(): String {
        return "Name: $employeeName Type: $employeeType ID: $employeeId"
    }
}

class Contracter(employeeName: String, employeeType: String, employeeId: Int) :
    Employee(employeeName, employeeType, employeeId) {
    var salary: String = "$100,000"
    override fun toString(): String {
        return super.toString() + " Salary: " + salary
    }
}

class Intern(employeeName: String, employeeType: String, employeeId: Int) :
    Employee(employeeName, employeeType, employeeId) {
    var salary: String = "$50,000"

    override fun toString(): String {
        return super.toString() + " Salary: " + salary
    }
}

class FullTime(employeeName: String, employeeType: String, employeeId: Int) :
    Employee(employeeName, employeeType, employeeId) {
    var salary: String = "$200,000"
    override fun toString(): String {
        return super.toString() + " Salary: " + salary
    }
}